### PR TITLE
Windows bruteforce analyzer

### LINKED ISSUE #2791: Windows brute force analyzer
Create a Windows login brute force analyzer that identifies a successful brute force login and tags the relevant events.

### PR DESCRIPTION
**Pull request by roshanmaskey**
This PR is related to issue # 2791

This PR adds support to determine successful windows login related to brute force activity. If a successful login is preceded by at least 20 failed login attempts from a given IP address within 3600 seconds, the analyzer will mark them as a brute force attempts.

**Checks**

- [x] All tests succeed.
- [x] Unit tests added.
- [x] e2e tests added.
- [ ] Documentation updated.

**Closing issues**

Put `closes #2791` in your comment to auto-close the issue that your PR fixes
(if such).

### CODE REVIEW

File: timesketch/lib/analyzers/authentication/windows.py [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+
+        if not event_ids:
+            log.debug("[%s] No events to annotate", self.NAME)
+            return
+
+        log.debug("[%s] Annotating %d events", self.NAME, len(event_ids))
+        for event in events:
+            if event.event_id in event_ids:
+                log.debug("[%s] Annotating event ID %s", self.NAME, event.event_id)
+                event.add_label("windows_bruteforce")
```
Comment (jkppr) [inline timesketch/lib/analyzers/authentication/windows.py:375]:
```suggestion
                event.add_tags(["windows_bruteforce"])
```
`add_label` is deprecated. Please use `add_tags` only.

**Review by jkppr — CHANGES_REQUESTED** [review]
Please check the comments below.

File: timesketch/lib/analyzers/authentication/windows.py [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+            event_data.source_port = event.source.get("port")
+            event_data.source_hostname = event.source.get("workstation_name")
+            event_data.session_id = event.source.get("logon_id")
+            event_data.eid = event.source.get("event_identifier")
+
+            # Handle events without logon_type
+            try:
+                event_data.logon_type = int(event.source.get("logon_type"))
+            except (ValueError, TypeError) as e:
+                log.warning("[%s] Unknown value for logon_type. %s", self.NAME, str(e))
```
[the remark this answers is not in the document]
Comment (roshanmaskey) [reply timesketch/lib/analyzers/authentication/windows.py]: Changed to debug.

File: timesketch/lib/analyzers/authentication/windows.py [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+
+        # Annotate the brute force events
+        events = self.event_stream(
+            query_string=self.SEARCH_QUERY, return_fields=self.EVENT_FIELDS
+        )
+
+        try:
+            self.annotate_events(events=events)
+        except (ValueError, TypeError) as e:
+            log.error("[%s] Unable to annotate. %s", self.NAME, str(e))
```
Comment (jkppr) [inline timesketch/lib/analyzers/authentication/windows.py:290]: This error is logged but seems to have no impact for the analyzer. If nothing should happen when this error is triggered, it is probably worth to change it to a debug level. If the error is important to the analyzer results, I would recommend to update the analyzer output object and set the results_status to ERROR and describe why this analyzer failed.

**Review by jkppr — CHANGES_REQUESTED** [review]

**Review by jkppr — APPROVED** [review]
lgtm

**Pull request merged**